src/middlewared/middlewared/alert/source/certificates.py

```diff
@@ -42,6 +42,56 @@ class CertificateIsExpiringAlertSource(AlertSource):
         return alerts
 
 
+class CertificateRevokedAlertClass(AlertClass):
+    category = AlertCategory.CERTIFICATES
+    level = AlertLevel.CRITICAL
+    title = 'Certificate Has Been Revoked'
+    text = '%(service)s %(type)s has been revoked.'
```

Comment:
```suggestion
    text = '%(service)s %(type)s has been revoked.'
```
What should the user do to fix the issue? Edit the revoked certificate, create a new one, etc ?

Reply:
This shows that the certificate is in use by one of the system's services and it should be replaced as soon as possible as it's been compromised and revoked.

Reply:
Updated suggestion to communicate the next step to the user.